src/openforms/submissions/views.py

```diff
@@ -116,3 +121,61 @@ def get_form_resume_url(self, submission: Submission) -> str:
         # Add the submission uuid to the query param
         form_resume_url.add({"submission_uuid": submission.uuid})
         return form_resume_url.url
+
+
+class SubmissionAttachmentDownloadView(LoginRequiredMixin, PrivateMediaView):
+    # only consider finished submissions (those are eligible for further processing)
+    # and submissions that have been successfully registered with the registration
+    # backend
+    queryset = SubmissionFileAttachment.objects.filter(
+        submission_step__submission__completed_on__isnull=False,
+        submission_step__submission__registration_status=RegistrationStatuses.success,
+    )
+    # expose UUIDs in URLs instead of enumeration-attack susceptible PKs
+    slug_field = "uuid"
+    slug_url_kwarg = "uuid"
+    file_field = "content"
+    permission_required = "submissions.view_submissionfileattachment"
+
+    def has_permission(self):
+        if not self.request.user.is_staff:
+            return False
+        return super().has_permission()
+
+    def get_object(self, queryset=None):
+        assert queryset is None, "Code path with explicit queryset not supported"
+
+        # check that a hash parameter is present on the request
+        content_hash = self.request.GET.get("hash")
+        if not content_hash:
+            logger.warning(
+                "Unauthorized file download attempted (missing content hash). Path: %s, user: %r",
+                self.request.path,
+                self.request.user,
+            )
+            raise PermissionDenied(_("File access denied."))
+
+        # cache the object for future lookups
+        if not hasattr(self, "_object"):
+            obj = super().get_object(queryset=queryset)
+
+            if not constant_time_compare(obj.content_hash, content_hash):
+                logger.warning(
+                    "Unauthorized file download attempted (invalid content hash). Path: %s, user: %r",
+                    self.request.path,
+                    self.request.user,
+                )
+                raise PermissionDenied(_("File access denied."))
```

Comment:
Does this redirect the user to the login screen or give a 403?

Reply:
403 - this is a codepath where people are tampering with hashes, so a harsher error is okay.

Additionally, the django `AccessMixin` only redirects if the user is not authenticated:

in `django.contrib.auth.mixins.AccessMixin.handle_no_permission`

```py
    def handle_no_permission(self):
        if self.raise_exception or self.request.user.is_authenticated:
            raise PermissionDenied(self.get_permission_denied_message())
```